core/src/main/java/info/rmapproject/core/model/impl/openrdf/OStatementsAdapter.java

```diff
@@ -469,29 +476,31 @@ else if (isDerivationEvent){
             }
             event = new ORMapEventDerivation(eventTypeStmt,eventTargetTypeStmt, associatedAgentStmt,
                     descriptionStmt, startTimeStmt,endTimeStmt, context, typeStatement, associatedKeyStmt,
-                    createdObjects,derivationStatement,sourceObjectStatement);
+                    lineageStatement, createdObjects,derivationStatement,sourceObjectStatement);
         }
         else if (isTombstoneEvent){
             if (tombstonedObjectStatement==null){
                 throw new RMapException("Tombstone event missing tombstoned object statement");
             }
             event = new ORMapEventTombstone(eventTypeStmt,eventTargetTypeStmt, associatedAgentStmt,
-                    descriptionStmt, startTimeStmt,endTimeStmt, context, typeStatement, associatedKeyStmt, tombstonedObjectStatement);
+                    descriptionStmt, startTimeStmt,endTimeStmt, context, typeStatement, associatedKeyStmt, 
+                    lineageStatement, tombstonedObjectStatement);
         }
         else if (isDeleteEvent){
             if (deletedObjectStatement==null){
                 throw new RMapException ("Delete event missing the deleted object statement");
             }
             event = new ORMapEventDeletion(eventTypeStmt,eventTargetTypeStmt, associatedAgentStmt,
-                    descriptionStmt, startTimeStmt,endTimeStmt, context, typeStatement, associatedKeyStmt, deletedObjectStatement);
+                    descriptionStmt, startTimeStmt,endTimeStmt, context, typeStatement, associatedKeyStmt, 
+                    lineageStatement, deletedObjectStatement);
         }
         else if (isReplaceEvent){
             if (replacedObjectStatement==null){
                 throw new RMapException("Update event missing replaced object statement");
             }
             event = new ORMapEventUpdateWithReplace(eventTypeStmt,eventTargetTypeStmt, associatedAgentStmt,
                     descriptionStmt, startTimeStmt, endTimeStmt, context, typeStatement, associatedKeyStmt,
-                    replacedObjectStatement);
+                    lineageStatement, replacedObjectStatement);
```

Comment:
One comment here - not sure what you want to do with this.  Replace Events only affect Agents.  Agents only ever have one ID, no lineage.  When an Agent goes through a Creation Event there is an `_:eventUri prov:generated _:agentUri` statement.  When an agent goes through an UpdateWithReplace Event, it has `_:eventUri rmap:updatedObject _:agentUri`. So that said, three choices... we remove the lineageProgenitor option for UpdateWithReplace Events, leave it like this but it's always null, or add it to all Events for consistency in categorization of Events.

Reply:
I like adding it to all events.  That's evolvable in the case that lineages are applicable to other things in the future, or new event types are added, that we want to associate with a lineage as well.